Resources/Views/index.leaf

```diff
@@ -256,6 +257,100 @@
+              <tt>dateFormatter.setLocalizedDateFormatFromTemplate(…)</tt>
+            </a> to account for the user's locale.
+          </li>
+          <li>When parsing ISO8601 internet dates, always use <tt>ISO8601DateFormatter</tt></li>
+          <li>
+            If you can't because your API format doesn't fit ISO8601 and you still absolutely need to use a custom <tt>dateFormat</tt>,
+            then be sure to also set your <tt>dateFormatter.locale</tt> to the special value <tt>Locale(identifier: "en_US_POSIX")</tt>.
+          </li>
+        </ul>
+
+        <h2> Formatting user-visible dates for your UI </h2>
+
+        <p class="quote">
+          Apple already has <a href="https://developer.apple.com/documentation/foundation/dateformatter#1680045">a dedicated paragraph</a>
+          in their documentation about best practices for formatting a date to present to the user in a locale-aware way.
+          Below is just the TL;DR.
+        </p>
+
+        <h3> Using <tt>dateStyle</tt> and <tt>timeStyle</tt> </h3>
+
+        <p>
+          The main recommendation to follow is to <strong>prefer using <tt>dateStyle</tt> and <tt>timeStyle</tt>
+          over <tt>dateFormat</tt></strong>.
+        </p><p>
+          This is because those are locale-aware and account for a lot of edge cases
+          that you can otherwise miss when using a custom and hardcoded <tt>dateFormat</tt>.
+        </p>
+        <p>
+          I would even advice you to convince your designer against using a custom format and explain them that Date
+          and Time is <a href="https://yourcalendricalfallacyis.com/">a tricky subject with lots of edge cases</a>
+          and that it's generally not worth using a custom format that might fit their assumptions about the locale
+          and region they're used to use but might not fit a lot of other's.
+        </p>
+
+        <h3> Using DateFormatter's templating methods to auto-adjust for the user's Locale </h3>
+
+        <p>If you still need to use a custom <tt>dateFormat</tt>, be sure that you use
+          <a href="https://developer.apple.com/documentation/foundation/dateformatter/1417087-setlocalizeddateformatfromtempla">
+            <tt>dateFormatter.setLocalizedDateFormatFromTemplate(…)</tt>
+          </a> or <a href="https://developer.apple.com/documentation/foundation/dateformatter/1408112-dateformat">
+            <tt>dateFormatter.dateFormat = DateFormatter.dateFormat(fromTemplate:…options:…locale:…)</tt>
+          </a> instead of setting it to a hard-coded <tt>String</tt>.
+        </p>
+
+        <h2> Working with ISO8601 dates and fixed formats for APIs </h2>
+
+        <p class="quote">
+            Apple Documentation <a href="https://developer.apple.com/documentation/foundation/dateformatter#2528261">
+            also has a dedicated section about those cases here</a>.
+        </p>
+
+        <h3> Using ISO8601DateFormatter </h3>
```

Comment:
My gripe about this class is that it doesn't handle milliseconds, which some frameworks produce in their "ISO8601" dates. I'm not sure if the standard is flexible on this or not, but Apple's class is not.

Reply:
Isn't it what [ISO8601DateFormatter.Options.withFractionalSeconds](https://developer.apple.com/documentation/foundation/iso8601dateformatter/options/2923300-withfractionalseconds) is for though?

Reply:
It does, actually. https://developer.apple.com/documentation/foundation/iso8601dateformatter/options/2923300-withfractionalseconds

The API is a bit esoteric since it's an option set. You have to call it like so `formatter.formatOptions.insert(. withFractionalSeconds)`

Reply:
Well TIL!  Thanks.

Man someone should create a website with all these tips...

Reply:
Added that tip in 0e108fb68c8ffd63d7e48824300724770acb5406